src/dune/process.ml

```diff
@@ -96,6 +96,26 @@ module Io = struct
     let channel = lazy (channel_of_descr (Lazy.force fd) mode) in
     { kind = File fn; mode; fd; channel; status = Close_after_exec }
 
+  include struct
+    let fname =
+      lazy
+        (Path.of_string
+           ( if Sys.win32 then
+             "nul"
+           else
+             "/dev/null" ))
+
+    let empty = lazy (Temp.create File ~prefix:"empty." ~suffix:".dune")
```

Comment:
Why not also use the null device for `stdin` ?

Reply:
Will it work on windows? If so, that's definitely better.

Reply:
Good question :) I will check later today and post back here.

Reply:
Seems to work in some cursory testing. But I don't know if there is some edge case where it doesn't, so perhaps using the temp file is better for extra safety.

@dra27 do you know if there is any edge case to be aware of when redirecting `stdin` to `nul` on Windows?

Reply:
It is notionally possible to disable the `NUL` DOS Device mapping (those mappings are in `HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\DOS Devices`) and I believe it is possible to disable `\??\Device\Null` device (which is `%SYSTEMROOT%\System32\drivers\null.sys`) from loading but both those of things are _really_ weird things to do! I'm fairly sure the CRT expects the NUL device to work.

I would expect you'd be far more likely to hit the obscure case of being unable to create the temporary the file than be unable to open `NUL`, so I'd definitely go with opening `Filename.null`. That said, are there Docker-like scenarios (not on Windows) where you might not have `/dev/null`? Windows DOS devices have the "benefit" that they're defined in every directory (like `.` and `..`).

Reply:
Thanks for the feedback. I switched to the `NUL` device for windows.